Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteAdapterProxy.cpp

```diff
@@ -60,13 +60,19 @@ void RemoteAdapterProxy::requestDevice(const PAL::WebGPU::DeviceDescriptor& desc
         return;
 
     auto [supportedFeatures, supportedLimits] = sendResult.takeReply();
+    if (!supportedLimits.maxTextureDimension2D) {
+        callback(nullptr);
+        return;
+    }
```

Comment:
Why is this limit special? This looks like it shouldn't be here.

Reply:
This limit is not special but it is to signify the device is not created via returning an empty value. We could check the entire structure, but it is not needed. If 0 is returned for something as basic as max 2d texture size, the device is not valid.

Reply:
I tried returning an optional<tuple<a, b>> from the IPC message instead of the tuple<a,b> that is currently returned but that became a lot of unnecessary changes and this approach is much simpler